packages/kbn-health-gateway-server/src/kibana/routes/root.ts

```diff
@@ -21,22 +20,24 @@ const GATEWAY_ROOT_ROUTE = '/';
 const KIBANA_ROOT_ROUTE = '/';
 
 interface RootRouteDependencies {
-  log: Logger;
+  logger: Logger;
   config: IConfigService;
 }
 
 type Fetch = (path: string) => Promise<Response>;
 
-export function createRootRoute({ config, log }: RootRouteDependencies) {
-  const kibanaConfig = new KibanaConfig(config.atPathSync<KibanaConfigType>('kibana'));
+export function createRootRoute({ config, logger }: RootRouteDependencies) {
+  const kibanaConfig = new KibanaConfig({ config, logger });
   const fetch = configureFetch(kibanaConfig);
 
   return {
     method: 'GET',
     path: GATEWAY_ROOT_ROUTE,
     handler: async (req: Request, h: ResponseToolkit) => {
-      const responses = await fetchKibanaRoots({ fetch, kibanaConfig, log });
+      const responses = await fetchKibanaRoots({ fetch, kibanaConfig, logger });
       const { body, statusCode } = mergeResponses(responses);
+      logger.debug(`Returning ${statusCode} response with body: ${JSON.stringify(body)}`);
```

Comment:
Main issue I see here is that we don't have anything to keep track of the execution context. The PR does add a sufficient amount of debug information ihmo, but we don't have anything to follow a specific request.

E.g if 2 requests hit this endpoint at the same time, given everything here is async, we won't be able to correlate the logs in the handler and in the `fetchKibanaRoots`  function (or eventually lower if the code becomes more complex at some point)

OTOH, I doubt we will be flooding this API, and I suspect that having the info that something went bad without knowing which specific request is occurred for is probably sufficient, so it's probably fine.

Reply:
Yeah, the way I think about it is:
- Control plane is going to be hitting the gateway regularly, but not so frequently that concurrent requests is going to be a realistic scenario in production
- If something goes wrong and an unhealthy response is returned, the container is going to be killed anyway
- If we are trying to analyze from logs why a container was killed, we are probably going to care less about the status code from the health gateway than we will about the responses/statuses from the upstream kibanas. We will want to know which Kibana process failed so we could look at those logs and attempt to figure out why.

With that in mind, I agree that what we have here is probably sufficient